selectors/emojis.ts

```diff
@@ -23,23 +22,42 @@ export const getEmojiMap = createSelector(
     },
 );
 
-export const getShortcutReactToLastPostEmittedFrom = (state: GlobalState) => state.views.emoji.shortcutReactToLastPostEmittedFrom;
+export const getShortcutReactToLastPostEmittedFrom = (state: GlobalState) =>
+    state.views.emoji.shortcutReactToLastPostEmittedFrom;
 
 export const getRecentEmojis = createSelector(
     'getRecentEmojis',
-    (state: GlobalState) => LocalStorageStore.getRecentEmojis(getCurrentUserId(state)),
-    (recentEmojis) => {
+
+    // if getSimplifiedData is true then return array of emoji names instead of RecentEmojiData object
+    (state: GlobalState, getSimplifiedData = true) => {
```

Comment:
This unfortunately won't work for a couple reasons.

First off, having this input selector return an object literal will mean that the final result function recalculates which would mean that a new array is returned each time this is called. Normally, you could solve that by just reverting this input selector and adding a separate one that just returns the argument to the result function, but since you could have two components calling this selector with a different value for `getSimplifiedData`, they'll end up overwriting each others memoized result and causing the same problem.

I think we're going to have to either split this into two selectors (maybe keep this and a `getRecentEmojiNames` to replace the existing one?) and/or update the other code that uses this to accept an array of emojis.